### PR TITLE
slightly clean up gpcopy test

### PR DESCRIPTION
**Pull request by d**
I was triggered when `gpcopy` was the only test that keeps failing on macOS, so I dug a little deeper, and found a few places that are not portable:

1. `echo -e` is a Bashism. (It fails on BSD and Debian-based systems that use `dash` as `sh`)
1. `exit -1` is also undefined behavior (It fails with `dash`)
1. "command not found" is uninteresting given our implementation. Worse, it's brittle.

### CODE REVIEW

**Comment by Eulerizeit** [comment]
Nice. I love little clean ups like this.

**Comment by hlinnaka** [comment]
+1
- Heikki

**Comment by d** [comment]
I'll back port to `5X_STABLE` once CI is green

File: src/test/regress/input/gpcopy.source [hunk 1]
```diff
@@ -1070,22 +1048,17 @@ DROP TABLE IF EXISTS COPY_TO_PROGRAM_ERROR;
 DROP TABLE IF EXISTS COPY_FROM_PROGRAM_ERROR;
 -- end_ignore
 CREATE TABLE COPY_TO_PROGRAM_ERROR(dir text);
-COPY COPY_TO_PROGRAM_ERROR TO PROGRAM 'ttt';
-\COPY COPY_TO_PROGRAM_ERROR TO PROGRAM 'ttt';
```
Comment (adam8157) [inline src/test/regress/input/gpcopy.source]: Better to get this `\COPY` case back.

**Comment by ashwinstar** [comment]
@d Please can we get this in.

**Comment by d** [comment]
Ah I had much less Github time in the past 1.5 week, sorry peeps. As Adam rightly mentioned, there was one legit coverage that I did remove from the `psql`'s backslash-copy that I should keep. I'll work on that

**Comment by d** [comment]
@adam8157 I tried retaining the negative test case of `\copy` but it still produces shell-dependent output. Namely, I got a diff like this on Debian:

```diff
*** /build/gpdb/src/test/regress/expected/gpcopy.out	Tue Nov 28 19:00:35 2017
--- /build/gpdb/src/test/regress/results/gpcopy.out	Tue Nov 28 19:00:35 2017
***************
*** 1183,1189 ****
  GP_IGNORE:-- end_ignore
  CREATE TABLE COPY_TO_PROGRAM_ERROR(dir text);
  \COPY COPY_TO_PROGRAM_ERROR TO PROGRAM 'ttt';
! sh: ttt: command not found
  ttt: command not found
  COPY COPY_TO_PROGRAM_ERROR TO PROGRAM 'echo && echo "error" >&2 && exit 255';
  ERROR:  command error message: error
--- 1217,1223 ----
  GP_IGNORE:-- end_ignore
  CREATE TABLE COPY_TO_PROGRAM_ERROR(dir text);
  \COPY COPY_TO_PROGRAM_ERROR TO PROGRAM 'ttt';
! sh: 1: ttt: not found
  ttt: command not found
  COPY COPY_TO_PROGRAM_ERROR TO PROGRAM 'echo && echo "error" >&2 && exit 255';
  ERROR:  command error message: error

======================================================================

```

This is the nature of `pg_regress` it seems, that we capture the stderr and stdout of `psql` and put them all into the output files...

**Comment by d** [comment]
Just rebased this PR. Note to reviewers: I gave up on the idea of keeping the negative "command not found" test case for `\copy` because it produces different outputs depending on the shell.

**Pull request merged**